### PR TITLE
DDP-8641 Patch for Singular Race Question Typo

### PR DESCRIPTION
**Pull request by MocanaAtBroad**
Just did a database update for this.
Put the expression in HOCON file instead of static variable in Java because I would have had to escape a lot of backslashes and would have gotten pretty ugly.
After applying patch, this is what race question looks like:

For child's survey:
Screen Shot 2022-08-30 at 4 21 58 PM

For the survey for self-enrollment:
Screen Shot 2022-08-30 at 4 23 59 PM

### CODE REVIEW

File: pepper-apis/studybuilder-cli/src/main/java/org/broadinstitute/ddp/studybuilder/task/SingularFixRaceQuestionTypo.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            "  join study_activity sa on sa.study_activity_id = q.study_activity_id " +
+            "  join i18n_template_substitution its on its.template_variable_id = tv.template_variable_id " +
+            "  join umbrella_study us on us.umbrella_study_id = sa.study_id " +
+            "set " +
+            "  substitution_value = ?" +
+            "where " +
+            "  tv.variable_name = ? " +
+            "  AND qsc.stable_id = ? " +
+            "  AND lc.iso_language_code = ? " +
+            "  AND us.guid = ?;";
```
Comment (bskinner) [inline pepper-apis/studybuilder-cli/src/main/java/org/broadinstitute/ddp/studybuilder/task/SingularFixRaceQuestionTypo.java]: The `+` operator needs to be on the new line, not on the end of the preceding one (this is a checkstyle error)

File: pepper-apis/studybuilder-cli/studies/singular/patches/singular-race-question-prompt-fix.conf [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "prompt": "What is $ddp.checkAnswer(\"PATIENT_SURVEY_WHO_ENROLLING_HIDDEN\",\"CHILD\", \"your child's\", \"your\") race?*"
```
Comment (bskinner) [inline pepper-apis/studybuilder-cli/studies/singular/patches/singular-race-question-prompt-fix.conf:2]: Are we still required to manually add a `*` after required questions or is the front-end handling that?

**Comment by ssettipalli** [comment]
I thought we can update the template text directly in i18n/en.conf and run the generic patch: UpdateTemplatesInPlace/UpdateActivityTemplatesInPlace to update templates.

File: pepper-apis/studybuilder-cli/src/main/java/org/broadinstitute/ddp/studybuilder/task/SingularFixRaceQuestionTypo.java (hunk 1 above)
Comment (MocanaAtBroad) [reply pepper-apis/studybuilder-cli/src/main/java/org/broadinstitute/ddp/studybuilder/task/SingularFixRaceQuestionTypo.java]: Oh geez! The IntelliJ was nice enough to convert my pasted string into multi-line concatenations for me.
I hope I don't mess it up trying to manually make checkstyle happy!

File: pepper-apis/studybuilder-cli/studies/singular/patches/singular-race-question-prompt-fix.conf (hunk 2 above)
Comment (MocanaAtBroad) [reply pepper-apis/studybuilder-cli/studies/singular/patches/singular-race-question-prompt-fix.conf:2]: I personally added the Angular code to add the asterisk years ago to required questions. I don't know what happened to it! But you see the screenshots (it was already in the original string we are editing).

**Comment by MocanaAtBroad** [comment]
> I thought we can update the template text directly in i18n/en.conf and run the generic patch: UpdateTemplatesInPlace/UpdateActivityTemplatesInPlace to update templates.

Oh! That probably would have been easier and cleaner. If you don't mind too much, let's just use this as is. I am hoping that are days of writing Java study config patches are coming to a close soon!

File: pepper-apis/studybuilder-cli/src/main/java/org/broadinstitute/ddp/studybuilder/task/SingularFixRaceQuestionTypo.java (hunk 1 above)
Comment (MocanaAtBroad) [reply pepper-apis/studybuilder-cli/src/main/java/org/broadinstitute/ddp/studybuilder/task/SingularFixRaceQuestionTypo.java]: Checkstyle issues fixed ffe2d5b76

File: pepper-apis/studybuilder-cli/studies/singular/patches/singular-race-question-prompt-fix.conf (hunk 2 above)
Comment (bskinner) [reply pepper-apis/studybuilder-cli/studies/singular/patches/singular-race-question-prompt-fix.conf:2]: I see what's going on here-

The question has a `required` validation, but the template is defined as
```
"""$tests_prompt_p1 <span class="Italic">$tests_prompt_p2</span>"""
```

So inserting a `*` at the end of the "prompt" would not result in the correct behavior.

File: pepper-apis/studybuilder-cli/src/main/java/org/broadinstitute/ddp/studybuilder/task/SingularFixRaceQuestionTypo.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            throw new DDPException("This task is only for the " + STUDY_GUID + " study!");
+        }
+    }
+
+    @Override
+    public void run(Handle handle) {
+        String newSubstitutionValue = dataCfg.getString("prompt");
+        int rowsModified = handle.execute(QUESTION_TEMPLATE_VARIABLE_SUBSTITUTION_VALUE, newSubstitutionValue,
+                TEMPLATE_VARIABLE_NAME, QUESTION_STABLE_ID, LANGUAGE_CODE_EN, STUDY_GUID);
+        log.info("Modified number of rows: {}", rowsModified);
```
Comment (bskinner) [inline pepper-apis/studybuilder-cli/src/main/java/org/broadinstitute/ddp/studybuilder/task/SingularFixRaceQuestionTypo.java:57]: Can we add a check to make sure that an expected number of updates is applied? Right now, there's no clear indicator to whomever is running the patch that it has been applied successfully.

File: pepper-apis/studybuilder-cli/studies/singular/patches/singular-race-question-prompt-fix.conf (hunk 2 above)
Comment (ssettipalli) [inline pepper-apis/studybuilder-cli/studies/singular/patches/singular-race-question-prompt-fix.conf:2]: @MocanaAtBroad 
Please apply this template change to i18n/en.conf too.
If not whenever we run UpdateTemplatesInPlace change will be lost.

Comment (MocanaAtBroad) [reply pepper-apis/studybuilder-cli/studies/singular/patches/singular-race-question-prompt-fix.conf:2]: 9d605b24f142b73ccba51322ac9ee81ae9b1ce45

**Review by ssettipalli — DISMISSED** [review]

**Pull request merged**